volmdlr/wires.py

```diff
@@ -1376,9 +1387,21 @@ def __init__(self, primitives: List[volmdlr.core.Primitive3D], color=None, alpha
         self.primitives = primitives
         self.color = color
         self.alpha = alpha
+        self.index = 0
         self.reference_path = reference_path
         PhysicalObject.__init__(self, name=name)
 
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.index >= len(self.primitives):
+            self.index = 0
+            raise StopIteration
+        else:
+            self.index += 1
+            return self.primitives[self.index-1]
```

Comment:
Because in the previous if it raises an error, you don't need to use an else statement here. You can delete the indentation of the block after it.